### PR TITLE
[Feature] support create/replace branch based on iceberg

### LINKED ISSUE #47846: Support iceberg timetravel/branch/tag
- [x] https://github.com/StarRocks/starrocks/pull/48061
- [x] https://github.com/StarRocks/starrocks/pull/47997
- [x] https://github.com/StarRocks/starrocks/pull/48063
- [x] https://github.com/StarRocks/starrocks/pull/47867
- [x] https://github.com/StarRocks/starrocks/pull/47847

## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

Management

- [ ]  Create/drop branch/tag
- [ ]  Write a table based on the branch
- [ ]  fast_forward branch to main
- [ ]  cherrypick_snapshot
- [ ]  Expire snapshot based on the ref options
- [ ] Auto expire snapshot(optional)

Query
- [ ]  Query with branch/tag
- [ ]  Query with version/timestamp
- [ ]  Query table snapshots/refs

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What I'm doing:
support creating/replacing branch syntax. Currently only implementing iceberg table
Fixes #issue
https://github.com/StarRocks/starrocks/issues/47846

syntax:
```
createReplaceBranchClause
    : (CREATE OR)? REPLACE BRANCH identifier branchOptions
    | CREATE BRANCH (IF NOT EXISTS)? identifier branchOptions
    ;
```

the simple demo based on latest snapshot with default properties
```
ALTER TABLE simple_demo CREATE BRANCH 
```

use all branch options demo:

```
ALTER TABLE test_table CREATE BRANCH test_branch 
AS OF VERSION 3144358988211265306   // based on which snapshot id
RETAIN 7 DAYS   // branch retention times
WITH SNAPSHOT RETENTION 5   // min retention snapshots
SNAPSHOTS 2 DAYS;  // snapshot max age on the branch
```
the refs contents in the iceberg metadata file of json.
```
"refs" : {
    "simple_demo" : {
      "snapshot-id" : 4842583333390156990,
      "type" : "branch"
    },
    "test_branch" : {
      "snapshot-id" : 3144358988211265306,
      "type" : "branch",
      "min-snapshots-to-keep" : 5,
      "max-snapshot-age-ms" : 172800000,
      "max-ref-age-ms" : 604800000
    },
    "main" : {
      "snapshot-id" : 4842583333390156990,
      "type" : "branch"
    }
  }
```

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/alter/AlterOpType.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public enum AlterOpType {
 
     // Optimize table
     OPTIMIZE,
-
+    ALTER_BRANCH,
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/alter/AlterOpType.java:69]: CREATE_BRANCH?

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -1312,6 +1316,42 @@ public ParseNode visitAlterTableStatement(StarRocksParser.AlterTableStatementCon
...
 
+    @Override
+    public ParseNode visitCreateOrReplaceBranchClause(StarRocksParser.CreateOrReplaceBranchClauseContext context) {
+        String branchName = getIdentifierName(context.identifier());
+
+        BranchOptions branchOptions = BranchOptions.empty();
+        if (context.branchOptions() != null) {
+            StarRocksParser.BranchOptionsContext branchOptionsContext = context.branchOptions();
+            Optional<Long> snapshotId =  Optional.ofNullable(branchOptionsContext.snapshotId())
+                    .map(id -> Long.parseLong(id.number().getText()));
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1327]: try catch Long.parseLong

**Comment by Youngwb** [comment]
I think there need a CreateBranchAnayzer to check the statement is legal and the value is valid, such as snapshot id, min-snapshots-to-keep, branch retention times value can used to execute

**Comment by stephen-shelby** [comment]
> I think there need a CreateBranchAnayzer to check the statement is legal and the value is valid, such as snapshot id, min-snapshots-to-keep, branch retention times value can used to execute

This is indeed my initial idea. However, this part has no illegal values and is relatively independent, put it here is easier to read.

File: fe/fe-core/src/main/java/com/starrocks/alter/AlterOpType.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/alter/AlterOpType.java:69]: there are create/replace/drop operations. maybe `alter` makes more sense

**Review by dirtysalt — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**